src/main/java/map/PathFindingService.java

```diff
@@ -25,36 +24,46 @@ public PathFindingService() { }
      * */
     public ArrayList<Node> genPath(MapNode start, MapNode dest, Boolean accessibility, String filter) {
 
-        AlgorithmContext ctx = new AlgorithmContext(new Astar());
         ArrayList<Node> target;
+        AlgorithmContext current;
 
         switch (filter) {
             case "astar":
-                ctx.setStrategy(new Astar());
-                target = ctx.findPathCTX(start, dest, accessibility, null);
-                estimatedTimeOfArrival = ctx.getEstimatedTime();
-                elevTimes = ctx.getElevTimes();
+                current = new Astar();
+                target = current.findDest(start, dest, accessibility, null);
+                estimatedTimeOfArrival = current.getEstimatedTime();
+                elevTimes = current.getElevTimes();
                 break;
             case "breadth":
-                ctx.setStrategy(new BreadthFS());
-                target = ctx.findPathCTX(start, dest, accessibility, null);
+                current = new BreadthFS();
+                target = current.findDest(start, dest, accessibility, null);
                 //estimatedTimeOfArrival = ctx.getEstimatedTime();
                 //elevTimes = ctx.getElevTimes();
                 break;
             case "depth":
-                ctx.setStrategy(new DepthFS());
-                target = ctx.findPathCTX(start, dest, accessibility, null);
-                estimatedTimeOfArrival = ctx.getEstimatedTime();
+                current = new DepthFS();
+                target = current.findDest(start, dest, accessibility, null);
+                estimatedTimeOfArrival = current.getEstimatedTime();
                 //elevTimes = ctx.getElevTimes();
                 break;
+            case "dijsktra":
+                current = new Dijsktra();
+                target = current.findDest(start, dest, accessibility, null);
+//                estimatedTimeOfArrival = current.getEstimatedTime();
+                break;
+            case  "best":
+                current = new BestFS();
+                target = current.findDest(start, dest, accessibility, null);
+                break;
             default:
-                ctx.setStrategy(new BreadthFS());
-                target = ctx.findPathCTX(start, null, accessibility, filter);
+                current = new BreadthFS();
```

Comment:
BFS should be its own switch case, ideally, default should never be called and is only there as a fail-safe